CI: Failing test on master

Seeing this on several PRs

```
___________________ TestTSPlot.test_mixed_freq_regular_first ___________________
[gw0] darwin -- Python 3.5.4 /Users/travis/miniconda3/envs/pandas/bin/python
self = <pandas.tests.plotting.test_datetimelike.TestTSPlot object at 0x10d3538d0>
    def test_mixed_freq_regular_first(self):
        # TODO
        s1 = tm.makeTimeSeries()
        s2 = s1[[0, 5, 10, 11, 12, 13, 14, 15]]
    
        # it works!
        s1.plot()
    
        ax2 = s2.plot(style='g')
        lines = ax2.get_lines()
        idx1 = PeriodIndex(lines[0].get_xdata())
        idx2 = PeriodIndex(lines[1].get_xdata())
>       assert idx1.equals(s1.index.to_period('B'))
```

<details>

```
E       AssertionError: assert False
E        +  where False = <bound method DatetimeIndexOpsMixin.equals of PeriodIndex(['2016-01-01 00:00', '2016-01-01 00:15', '2016-01-01 00:30',\n             '2016-01-01 00:45'],\n            dtype='period[T]', freq='T')>(PeriodIndex(['2000-01-03', '2000-01-04', '2000-01-05', '2000-01-06',\n             '2000-01-07', '2000-01-10', '2000-01...-05-15',\n             '2000-05-16', '2000-05-17', '2000-05-18', '2000-05-19'],\n            dtype='period[B]', freq='B'))
E        +    where <bound method DatetimeIndexOpsMixin.equals of PeriodIndex(['2016-01-01 00:00', '2016-01-01 00:15', '2016-01-01 00:30',\n             '2016-01-01 00:45'],\n            dtype='period[T]', freq='T')> = PeriodIndex(['2016-01-01 00:00', '2016-01-01 00:15', '2016-01-01 00:30',\n             '2016-01-01 00:45'],\n            dtype='period[T]', freq='T').equals
E        +    and   PeriodIndex(['2000-01-03', '2000-01-04', '2000-01-05', '2000-01-06',\n             '2000-01-07', '2000-01-10', '2000-01...-05-15',\n             '2000-05-16', '2000-05-17', '2000-05-18', '2000-05-19'],\n            dtype='period[B]', freq='B') = <bound method DatetimeIndex.to_period of DatetimeIndex(['2000-01-03', '2000-01-04', '2000-01-05', '2000-01-06',\n      ...              '2000-05-16', '2000-05-17', '2000-05-18', '2000-05-19'],\n              dtype='datetime64[ns]', freq='B')>('B')
E        +      where <bound method DatetimeIndex.to_period of DatetimeIndex(['2000-01-03', '2000-01-04', '2000-01-05', '2000-01-06',\n      ...              '2000-05-16', '2000-05-17', '2000-05-18', '2000-05-19'],\n              dtype='datetime64[ns]', freq='B')> = DatetimeIndex(['2000-01-03', '2000-01-04', '2000-01-05', '2000-01-06',\n               '2000-01-07', '2000-01-10', '200...               '2000-05-16', '2000-05-17', '2000-05-18', '2000-05-19'],\n              dtype='datetime64[ns]', freq='B').to_period
E        +        where DatetimeIndex(['2000-01-03', '2000-01-04', '2000-01-05', '2000-01-06',\n               '2000-01-07', '2000-01-10', '200...               '2000-05-16', '2000-05-17', '2000-05-18', '2000-05-19'],\n              dtype='datetime64[ns]', freq='B') = 2000-01-03    0.430091\n2000-01-04    1.453439\n2000-01-05   -0.202244\n2000-01-06    0.168447\n2000-01-07   -0.800664\n200...6    0.565288\n2000-05-17    1.618565\n2000-05-18   -0.774342\n2000-05-19   -0.747118\nFreq: B, Length: 100, dtype: float64.index
pandas/tests/plotting/test_datetimelike.py:698: AssertionError
```

</details>

I wasn't able to quickly reproduce it with MPL 2.1.2 and NumPy 1.10.4; I'll look into it more in ~1 hour.

--- a/pandas/tests/plotting/test_datetimelike.py
+++ b/pandas/tests/plotting/test_datetimelike.py
@@ -689,14 +689,17 @@ class TestTSPlot(TestPlotBase):
         s2 = s1[[0, 5, 10, 11, 12, 13, 14, 15]]
 
         # it works!
-        s1.plot()
+        _, ax = self.plt.subplots()
+        s1.plot(ax=ax)
 
-        ax2 = s2.plot(style='g')
+        ax2 = s2.plot(style='g', ax=ax)
         lines = ax2.get_lines()
         idx1 = PeriodIndex(lines[0].get_xdata())
         idx2 = PeriodIndex(lines[1].get_xdata())
-        assert idx1.equals(s1.index.to_period('B'))
-        assert idx2.equals(s2.index.to_period('B'))
+
+        tm.assert_index_equal(idx1, s1.index.to_period('B'))
+        tm.assert_index_equal(idx2, s2.index.to_period('B'))
+
         left, right = ax2.get_xlim()
         pidx = s1.index.to_period()
         assert left <= pidx[0].ordinal

